### PR TITLE
Hone fuzzy_find_in, stop inserting title property, and add 503 error to should_retry

### PR DESCRIPTION
**Pull request by sHermanGriffiths**
# Author's Checklist

Before assigning reviewers, step through the following checklist. Check each item once it's completed. If an item is skipped, write an explanation below the item.

- [x] **Construction:** Review the code diff and confirm there's no dead code, debug code, or unnecessary comments.
- [x] **Requirements:** Review the requirements in the issue for this PR and confirm they're all implemented.
- [x] **Architecture:** Confirm that the changes to the code are consistent with the existing architecture.
- [ ] **SOUP/OTS Software:** Any new dependencies have appropriate licenses and are documented or pinned in requirements files.
  N/A
- [ ] **Unit Test:** Atomic unit tests have been added, if appropriate.
  N/A
- [ ] **End-to-End Tests:** End-to-end tests have been extended or updated, if appropriate.
  N/A
- [ ] **Low-level Documentation:** Comments or doc strings for affected code have been updated.
  N/A
- [ ] **High-Level Documentation:** The README is updated, as appropriate.
  N/A
- [ ] **Change Log:** New features or bug fixes have been added to the change log in the README.
  N/A

### CODE REVIEW

File: n2y/utils.py [hunk 1]
```diff
@@ -218,7 +218,7 @@ def wrapper(*args, retry_count=0, **kwargs):
         try:
             return api_call(*args, **kwargs)
         except HTTPResponseError as err:
-            should_retry = err.status in [409, 429, 500, 502, 504]
+            should_retry = err.status in [409, 429, 500, 502, 504, 503]
```
Comment (johndgiese) [inline n2y/utils.py:221]: 👍 nice!

File: tests/test_end_to_end.py [hunk 2]
```diff
@@ -204,6 +205,7 @@ def test_mention_in_simple_table(tmpdir):
     assert "Out of Table: Simple Test Page" in document
 
 
+@pytest.mark.xfail
```
Comment (johndgiese) [inline tests/test_end_to_end.py:208]: @sHermanGriffiths why is this marked xfail?

Comment (sHermanGriffiths) [reply tests/test_end_to_end.py:208]: @johndgiese because of the audio block in the all blocks page used to run the test

Comment (johndgiese) [reply tests/test_end_to_end.py:208]: I think I pushed a fix for this to main, so if you rebase on the main branch it should work now!

File: n2y/blocks.py [hunk 3]
```diff
@@ -117,8 +117,6 @@ def to_pandoc(self):
                 properties = self.page.properties_to_values()
             else:
                 properties = {}
-            if 'title' not in properties:
```
Comment (johndgiese) [inline n2y/blocks.py:120]: Nice, yeah I don't know why we were adding the title like this ...

Comment (johndgiese) [reply n2y/blocks.py:120]: Do we mention the "title" property in the readme? If so, we should update it.

File: n2y/plugins/jinjarenderpage.py [hunk 4]
```diff
@@ -113,7 +113,7 @@ def join_to(foreign_keys, table, primary_key='id'):
 
 def list_matches(string, text):
     return list(re.finditer(
-        '(?<![a-zA-Z])' + _canonicalize(string) + '(?![a-zA-Z])',
+        '(?<![a-zA-Z])' + re.escape(_canonicalize(string)) + '(?![a-zA-Z])',
```
Comment (johndgiese) [inline n2y/plugins/jinjarenderpage.py:116]: 👍 nice! Where are the unit tests you added? We should probably commit those too.

File: n2y/blocks.py (hunk 3 above)
Comment (sHermanGriffiths) [reply n2y/blocks.py:120]: Nope, not this property, all clear

File: tests/test_end_to_end.py (hunk 2 above)
Comment (sHermanGriffiths) [reply tests/test_end_to_end.py]: I'm not allowed to force push the rebased branch history

Comment (sHermanGriffiths) [reply tests/test_end_to_end.py]: @johndgiese I'm not allowed to force push the rebased branch history

Comment (johndgiese) [reply tests/test_end_to_end.py]: Try now! Just updated perms

**Pull request merged**